src/components/ProductList.tsx

```diff
@@ -0,0 +1,135 @@
+import { SkeletonProduct } from '@/components/Skeleton';
+import { unstable_noStore as noStore } from 'next/cache';
+
+const ProductList = () => {
+  const { data, isLoading, error } = useQuery<any>({
+    queryKey: ['products'],
+    queryFn: async () => {
+      noStore();
+      const response: ProductObj= await request.get(
+        `/products?page=${1}&limit=${10}`,
+      );
+      console.log(response);
+      const data = response.products;
+
+      return data;
+    },
+  });
+
+  if (isLoading) {
+    return (
+      <div>
+        <SkeletonProduct />
+      </div>
+    );
+  }
+
+  if (error) {
+    return <div>Error: {error.message}</div>;
+  }
+
+  if (!data) {
+    return <div>No products found.</div>;
+  }
+
+  return (
+    <>
+      <div className="mt-5 pl-20 w-full">
+        {data && (
+          <>
+            <ul className="ml-2 flex flex-row justify-between flex-wrap items-center">
+              {data.map(
+                (
+                  product: {
+                    id: string;
+                    productPrice: number;
+                    productThumbnail: string;
+                    productDescription: string;
+                  },
+                  i: number,
+                ) => (
+                  <Card
+                    key={i.toString()}
+                    id={product.id}
+                    productPrice={product.productPrice}
+                    productThumbnail={product.productThumbnail}
+                    productDescription={product.productDescription}
+                  />
+                ),
+              )}
+            </ul>
+          </>
+        )}
+        <nav aria-label="Page navigation example" className="mt-3">
+          <ul className="w-full inline-flex -space-x-px text-sm justify-center">
+            <li>
+              <a
+                href="#"
+                className="flex items-center justify-center px-3 h-8 ms-0 leading-tight text-gray-500 bg-white border border-e-0 border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
+              >
+                Previous
+              </a>
+            </li>
+            <li>
+              <a
+                href="#"
+                className="flex items-center justify-center px-3 h-8 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
+              >
+                1
+              </a>
+            </li>
+            <li>
+              <a
+                href="#"
+                className="flex items-center justify-center px-3 h-8 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
```

Comment:
Why are all these codes duplicated? You can put them in loop if they have similar properties.